cli-bss.py - Argument --offset

emmanuel-benoit opened the pull request:
Ajouté l'argument --offset pour les listes de comptes ou de CoS.

emmanuel-benoit committed:
cli-bss.py - Argument --offset

Ajouté l'argument --offset pour les listes de comptes ou de CoS.

salaun-urennes1 commented on cli-bss.py:
```diff
@@ -352,10 +357,14 @@
         print(cos.showAttr())
 
 elif args['getAllCos'] == True:
+    action_args = {
```
Une typo dans le code proposée : les clés du dictionaire action_args doivent être entre quotes. Sinon : 
```
Traceback (most recent call last):
  File "./cli-bss.py", line 361, in <module>
    domain: args[ 'domain' ] ,
NameError: name 'domain' is not defined
```

emmanuel-benoit replied:
Corrigé.

salaun-urennes1 replied:
Bonjour Emmanuel,
Merci pour le correctif ; le code s'exécute maintenant.
Par contre il semble que l'API BSS ne supporte pas les paramètres limit et offset pour la fonction GetAllCos, cf https://dashboard.partage.renater.fr/api_bss_documentation.html#_6_1_getallcos.

J'ai effectué un test qui semble confirmer qu'il ignore ces paramètres :
`./cli-bss.py --domain=univ-rennes1.fr --domainKey=XXX --getAllCos --limit=1 --offset=1` me retourne 4 résultats, en partant du 1er.

emmanuel-benoit replied:
Bonjour Olivier.
OK. Le code d'origine avait le paramètre `limit`, j'ai donc rajouté l'offset sans vérifier (nous n'avons que 3 CoS...).
Veux-tu que je supprime carrément l'utilisation de ces paramètres pour `GetAllCos`?

salaun-urennes1 replied:
Oui, s'il te plait.

emmanuel-benoit replied:
OK. J'ai aussi supprimé le paramètre ldap_query qui n'était pas pris en compte non plus.

salaun-urennes1 reviewed (changes requested):
Def de action_args à corriger

emmanuel-benoit committed:
cli-bss - Rajouté délimiteurs de chaînes manquants

emmanuel-benoit committed:
services.COSService.getAllCOS() - Supprimé paramètres inutiles

Les paramètres limit, offset et ldap_query ne servent en réalité à rien
sur l'appel GetAllCos de l'API Partage. Ils ont été supprimés de la
fonction du service, ainsi que du client en ligne de commande.
